src/components/Tasks.tsx

```diff
@@ -2,56 +2,159 @@ import { type Task as TaskType } from "./TaskInterface";
+    if (tasks.length > 0) {
+      localStorage.setItem("tasks", JSON.stringify(tasks));
+    }
   }, [tasks]);
 
-  const createItem = (task: string, category: string, priority: string) => {
-    let completed = false;
+  const createItem = (task: string, category: string, isUrgent: boolean) => {
+    const existingTask = tasks.find((element) => element.task === task);
 
-    const markCompleted = () => {
-      setTasks((prevTasks) =>
-        prevTasks.map((prevTask) =>
-          prevTask.task === task ? { ...prevTask, isCompleted: true } : prevTask
-        )
-      );
-    };
+    if (existingTask) {
+      return;
+    }
 
-    const deleteTask = () => {
-      setTasks(tasks.filter((element) => element.task !== task));
-    };
+    let completed = false;
 
     setTasks((prev) => [
       ...prev,
       {
         isCompleted: completed,
         category,
         task,
-        priority,
-        markCompleted,
-        deleteTask,
+        isUrgent,
       },
     ]);
   };
 
-  const displayTasks = tasks.map((task) => {
-    return <Task key={task.task} {...task}></Task>;
-  });
+  const filterTasks = (isUrgent: boolean) => (category: string) => {
+    return tasks.filter(
+      (task) => task.isUrgent === isUrgent && task.category === category
+    );
+  };
+
+  //not the best example of currying, couldn't think of anything better
+  const displayTasks =
+    isUrgent && selectedCategory
+      ? filterTasks(isUrgent)(selectedCategory).map((item) => {
+          return (
+            <Task
+              key={item.task}
+              {...item}
+              markCompleted={markCompleted}
+              deleteTask={deleteTask}
+            ></Task>
+          );
+        })
+      : tasks.map((item) => {
+          return (
+            <Task
+              key={item.task}
+              {...item}
+              markCompleted={markCompleted}
+              deleteTask={deleteTask}
+            ></Task>
+          );
+        });
 
   return (
     <div className="tasks">
-      <button
-        onClick={() => {
-          createItem("mytask", "mycategory", "urgent");
-        }}
-        className="green"
-      >
-        create
-      </button>
+      {!openCreateTask && (
+        <button
+          onClick={() => {
+            setOpenCreateTask((prev) => !prev);
+            setIsUrgent((prev) => !prev);
+            setSelectedCategory("");
+          }}
+          className="green"
+        >
+          add
+        </button>
+      )}
+      {openCreateTask && (
+        <CreateTask
+          categories={categories}
+          createItem={createItem}
+          close={() => {
+            setOpenCreateTask(false);
+          }}
+        ></CreateTask>
+      )}
+
+      {!openCreateTask && (
+        <div className="filter">
+          <div>
+            <p>filter urgent tasks:</p>
+            <button
+              style={
+                isUrgent ? { backgroundColor: "#9e2b2b", color: "white" } : {}
```

Comment:
Usually, react developers apply classes with predefined styles instead of using inline styles.
For the purpose of this repository, this is OK.